src/text/parsers/top_level.rs

```diff
@@ -0,0 +1,42 @@
+use nom::branch::alt;
+use nom::combinator::opt;
+use nom::sequence::preceded;
+use nom::IResult;
+
+use crate::text::parsers::blob::parse_blob;
+use crate::text::parsers::boolean::parse_boolean;
+use crate::text::parsers::clob::parse_clob;
+use crate::text::parsers::containers::{parse_container_end, parse_container_start};
+use crate::text::parsers::decimal::parse_decimal;
+use crate::text::parsers::float::parse_float;
+use crate::text::parsers::integer::parse_integer;
+use crate::text::parsers::null::parse_null;
+use crate::text::parsers::string::parse_string;
+use crate::text::parsers::symbol::parse_symbol;
+use crate::text::parsers::timestamp::parse_timestamp;
+use crate::text::parsers::whitespace;
+use crate::text::TextStreamItem;
```

Comment:
The rules defined in this file originally lived alongside the `TextReader` in `text::reader.rs`. They've been promoted to their own module.